Apps/Sandcastle/gallery/Animation Speed.html

```diff
@@ -0,0 +1,115 @@
+    <meta name="description" content="Control glTF animation speed based on the entity's velocity.">
+    <meta name="cesium-sandcastle-labels" content="Showcases">
+    <title>Cesium Demo</title>
+    <script type="text/javascript" src="../Sandcastle-header.js"></script>
+    <script type="text/javascript" src="../../../ThirdParty/requirejs-2.1.20/require.js"></script>
+    <script type="text/javascript">
+    require.config({
+        baseUrl : '../../../Source',
+        waitSeconds : 60
+    });
+    </script>
+</head>
+<body class="sandcastle-loading" data-sandcastle-bucket="bucket-requirejs.html">
+<style>
+    @import url(../templates/bucket.css);
+</style>
+<div id="cesiumContainer" class="fullSize"></div>
+<div id="loadingOverlay"><h1>Loading...</h1></div>
+<div id="toolbar"></div>
+<script id="cesium_sandcastle_script">
+function startup(Cesium) {
+    'use strict';
+//Sandcastle_Begin
+var viewer = new Cesium.Viewer('cesiumContainer', {
+    shouldAnimate: true
+});
+
+//Make sure viewer is at the desired time.
+var start = Cesium.JulianDate.fromDate(new Date(2018, 11, 12));
+var totalSeconds = 30;
+var stop = Cesium.JulianDate.addSeconds(start, totalSeconds, new Cesium.JulianDate());
+viewer.clock.startTime = start.clone();
+viewer.clock.stopTime = stop.clone();
+viewer.clock.currentTime = start.clone();
+viewer.clock.clockRange = Cesium.ClockRange.LOOP_STOP; //Loop at the end
+viewer.timeline.zoomTo(start, stop);
+
+// Create a path from a set of positions for our vehicle.
+var position = new Cesium.SampledPositionProperty();
+// These numbers were obtained by starting with 3 Cartesian3 locations, and then using
+// Cesium.Cartesian3.lerp to create more positions between them. They were constructed
+// so that the first few are close together and the last two are far apart to show
+// how the animation picks up as the vehicle's speed changes.
+var locations = [
+    new Cesium.Cartesian3(-2379510.08905552,-4665419.64840452,3628182.20006795),
+    new Cesium.Cartesian3(-2379521.766634856, -4665446.787560957, 3628139.926700882),
+    new Cesium.Cartesian3(-2379533.4442141918, -4665473.926717391, 3628097.653333814),
+    new Cesium.Cartesian3(-2379545.121793528, -4665501.065873828, 3628055.379966746),
+    new Cesium.Cartesian3(-2379556.799372864, -4665528.205030263, 3628013.106599678),
+    new Cesium.Cartesian3(-2379568.4769522,-4665555.3441867,3627970.83323261),
+    new Cesium.Cartesian3(-2379603.7074103747, -4665623.48990283, 3627860.82704567)
+];
+
+var stepSize = totalSeconds / (locations.length - 1);
+
+for (var i = 0; i < locations.length; ++i) {
+    var location = locations[i];
+    var time = Cesium.JulianDate.addSeconds(start, i * stepSize, new Cesium.JulianDate());
+    position.addSample(time, location);
+}
+
+// A velocity vector property will give us the entity's speed and direction at any given time.
```

Comment:
I would move these variables to right before the function that uses them, so that it's more clear to the reader.